Framework/PythonInterface/mantid/plots/__init__.py

```diff
@@ -1045,6 +1048,217 @@ def tricontourf(self, *args, **kwargs):
         """
         return self._plot_2d_func('tricontourf', *args, **kwargs)
 
+    def set_initial_dimensions(self):
+        # Set the width and height which are used to calculate the offset percentage for waterfall plots.
+        # This means that the curves in a waterfall plot are always offset by the same amount, even if the
+        # plot limits change.
+        x_lim, y_lim = self.get_xlim(), self.get_ylim()
+        self.width = x_lim[1] - x_lim[0]
+        self.height = y_lim[1] - y_lim[0]
+
+    def is_waterfall_plot(self):
+        return self.waterfall_x_offset != 0 or self.waterfall_y_offset != 0
+
+    def update_waterfall_plot(self, x_offset, y_offset):
+        errorbar_cap_lines = self.remove_and_return_errorbar_cap_lines()
+
+        for i in range(len(self.get_lines())):
+            self.convert_single_line_to_waterfall(i, x_offset, y_offset)
+
+        # Now set the new offsets
+        self.waterfall_x_offset = x_offset
+        self.waterfall_y_offset = y_offset
+
+        if any(isinstance(collection, PolyCollection) for collection in self.collections):
+            self.waterfall_update_fill()
+
+        self.lines += errorbar_cap_lines
+
+        self.set_waterfall_toolbar_options_enabled()
+        self.get_figure().canvas.draw()
+
+    def set_waterfall_toolbar_options_enabled(self):
+        toolbar = self.get_figure().canvas.toolbar
+        toolbar.waterfall_conversion(self.is_waterfall_plot())
+
+    def convert_to_waterfall(self):
```

Comment:
The matplotlib api tends to go with `set_` and then a bool, e.g. Legend.set_draggable. Can we follow this with `set_waterfall` that accepts a single `state` parameter? I wouldn't be so bothered but these will be user-facing functions when they are used in scripts so we need to make sure they are consistent.